collectors/30/yarn_mapreduce.py

```diff
@@ -17,7 +25,11 @@
 
 
 def main():
-    running_apps = get_json(APP_BASE_URL + '?state=RUNNING')
+    try:
+        running_apps = get_json(APP_BASE_URL + '?state=RUNNING')
+    except Exception, e:
+        # This will fail if not the active RM so let's exit gracefully so we don't mark this collector as dead
```

Comment:
Can we log anything here?

Reply:
get_json already has pretty verbose logging.